packages/core/src/lib/createSystem.ts

```diff
@@ -107,15 +107,19 @@ export function createSystem(config: KeystoneConfig) {
     graphQLSchema,
     adminMeta,
     getKeystone: (prismaModule: PrismaModule) => {
-      const prePrismaClient = new prismaModule.PrismaClient({
+      const clientConfig: Parameters<NonNullable<KeystoneConfig['db']['prismaClient']>>[0] = {
         datasources: { [config.db.provider]: { url: config.db.url } },
         log:
           config.db.enableLogging === true
             ? ['query']
             : config.db.enableLogging === false
             ? undefined
             : config.db.enableLogging,
-      });
+      };
+
+      const prePrismaClient = config.db.prismaClient
+        ? config.db.prismaClient(clientConfig)
```

Comment:
theoretically we could do `instanceof prismaModule.PrismaClient` 🤔 this should probably have the correct instance of the prisma client which the end user is using

Reply:
I don't think we need to do that, we are relying on Typescript in many scenarios, I don't think we need to introduce a run-time burden for this.

I am open to adding something like `zod` for the configuration at some point, but, Typescript is the priority.